File: WorldConquest/src/models/Building.ts
// Building Model

/**
 * Represents a building in a province
 */
export class Building {
    // Building ID
    id: string;
    
    // Building name
    name: string;
    
    // Building type
    type: BuildingType;
    
    // Building level (for upgradable buildings)
    level: number;
    
    // Construction cost
    cost: number;
    
    // Maintenance cost per turn
    maintenance: number;
    
    // Income generated per turn
    income: number;
    
    // Effects provided by the building
    effects: BuildingEffects;
    
    constructor(id: string, name: string, type: BuildingType, cost: number, maintenance: number = 0, income: number = 0) {
        this.id = id;
        this.name = name;
        this.type = type;
        this.level = 1;
        this.cost = cost;
        this.maintenance = maintenance;
        this.income = income;
        this.effects = new BuildingEffects();
    }
}

// Building type enum
export enum BuildingType {
    Economic,
    Military,
    Administrative,
    Religious,
    Special
}

// Building effects class
export class BuildingEffects {
    // Economic effects
    taxModifier: number = 0;
    productionModifier: number = 0;
    
    // Military effects
    garrisonBonus: number = 0;
    recruitmentBonus: number = 0;
    
    // Administrative effects
    unrestReduction: number = 0;
}

File: WorldConquest/Models/Building.cs
using System.Text.Json.Serialization;

namespace WorldConquest.Models
{
    public class Building
    {
        // Building ID
        public string Id { get; set; }
        
        // Building name
        public string Name { get; set; }
        
        // Building type
        public BuildingType Type { get; set; }
        
        // Building level (for upgradable buildings)
        public int Level { get; set; }
        
        // Construction cost
        public int Cost { get; set; }
        
        // Maintenance cost per turn
        public int Maintenance { get; set; }
        
        // Income generated per turn
        public int Income { get; set; }
        
        // Effects provided by the building
        public BuildingEffects Effects { get; set; }
        
        public Building()
        {
            // Default constructor for serialization
        }
        
        public Building(string id, string name, BuildingType type, int cost, int maintenance = 0, int income = 0)
        {
            Id = id;
            Name = name;
            Type = type;
            Level = 1;
            Cost = cost;
            Maintenance = maintenance;
            Income = income;
            Effects = new BuildingEffects();
        }
    }
    
    public enum BuildingType
    {
        Economic,
        Military,
        Administrative,
        Religious,
        Special
    }
    
    public class BuildingEffects
    {
        // Economic effects
        public int TaxModifier { get; set; } = 0;
        public int ProductionModifier { get; set; } = 0;
        
        // Military effects
        public int GarrisonBonus { get; set; } = 0;
        public int RecruitmentBonus { get; set; } = 0;
        
        // Administrative effects
        public int UnrestReduction { get; set; } = 0;
        
        // Other effects can be added as needed
    }
}